File: Python/Programmers/Level 2/h-index.py
def solution(citations):
    citations.sort()
    answer = 0
    for i in range(len(citations)):
        h = citations[i]
        if len(citations[i:]) <= h < len(citations[:i + 1]):
            answer = h
            break

    return answer


citations = [10, 11, 12, 13, 14, 15, 16, 17, 18, 19]
print(solution(citations))

File: Java/src/Programmers/Sort/HIndex.java
package Programmers.Sort;

import java.util.Arrays;

public class HIndex {
    public int solution(int[] citations) {
        int answer = 0;
        int n = citations.length;

        Arrays.sort(citations);

        for (int i = 0; i < n; i++){
            int h = citations.length - i;
            if(h <= citations[i]) {
                answer = h;
                break;
            }
        }

        return answer;
    }

    public static void main(String[] args) {
//        int [] citations = {3, 0, 6, 1, 5};
        int [] citations = {10, 55, 100};
        HIndex solution = new HIndex();
        System.out.println(solution.solution(citations));
    }
}
